modules/iam/variables.tf

```diff
@@ -0,0 +1,160 @@
+variable "name_prefix" {
+  description = "Prefix used in names for the resources. (IAM Role, Instance Profile)"
+  type        = string
+}
+
+variable "tags" {
+  description = "Global tags configured for all provisioned resources."
+  type        = map(any)
+}
+
+variable "region" {
+  description = "AWS region where SSM or CloudWatch is located."
+  type        = string
+  default     = null
+}
+
+variable "principal_role" {
+  description = "The type of entity that can take actions in AWS."
+  type        = string
+  default     = "ec2.amazonaws.com"
+}
+
+variable "create_role" {
+  description = "Create a dedicated role creation of pre-defined policies."
+  type        = bool
+  default     = true
+}
+
+variable "role_name" {
+  description = "A role name, required for the service."
+  type        = string
+}
+
+variable "create_instance_profile" {
+  description = "Create an instance profile."
+  type        = bool
+  default     = false
+}
+
+variable "instance_profile_name" {
+  description = "Instance profile name."
+  type        = string
+  default     = null
+}
+
+variable "create_lambda_policy" {
+  description = "Create a pre-defined lambda policies for ASG."
+  type        = bool
+  default     = false
+}
+
+variable "create_vmseries_policy" {
+  description = "Create a pre-defined vmseries policy."
+  type        = bool
+  default     = false
+}
+
+variable "create_bootrap_policy" {
+  description = "Create a pre-defined bootstrap policy."
+  type        = bool
+  default     = false
+}
+
+variable "create_panorama_policy" {
+  description = "Create a pre-defined panorama policy."
+  type        = bool
+  default     = false
+}
+
+variable "policy_arn" {
+  description = "The AWS or Customer managed policy arn. It should be used for spoke VM scenario using the AWS managed ```arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore``` policy."
+  type        = string
+  default     = null
+}
+
+variable "custom_policy" {
+  description = <<-EOF
+  A custom lambda policy. Multi-statement is supported.
```

Comment:
Does custom policy applies to labda only? Can't user have a custom policy for VM-Series or other resources created with the code?